pubsub/utils.go

```diff
@@ -0,0 +1,69 @@
+package pubsub
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Fetches system certs and returns them if possible. If unable to fetch system certs then an empty cert pool is returned instead.
+func getCerts() *x509.CertPool {
+	if certs, err := x509.SystemCertPool(); err == nil {
+		return certs
+	}
+
+	return x509.NewCertPool()
+}
+
+// parseUnionFields parses the schema JSON to identify avro union fields.
+func parseUnionFields(_ context.Context, schemaJSON string) (map[string]struct{}, error) {
+	var schema map[string]interface{}
+	if err := json.Unmarshal([]byte(schemaJSON), &schema); err != nil {
+		return nil, fmt.Errorf("failed to parse schema: %w", err)
+	}
+
+	unionFields := make(map[string]struct{})
+	fields := schema["fields"].([]interface{})
```

Comment:
are these castings safe? if not, this will cause a panic.
should be something like `if fields, ok := schema["fields"].([]interface{}); ok {// do sth}`

Reply:
Updated! :D Lmk if it makes sense!